47.md

```diff
@@ -63,6 +63,269 @@ Response:
+        ]
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "pay_keysend",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "preimage": "0123456789abcdef...", // preimage of the payment
+        "payment_hash": "31afdf1..", // payment hash of the payment
+    }
+}
+```
+
+### Make Invoice
+
+Request:
+```jsonc
+{
+    "method": "make_invoice",
+    "params": {
+        "amount": 123, // invoice amount in msats, required
+        "description": "some description", // invoice description, optional
+        "description_hash": "31afdf1..", // invoice description hash, optional
+        "preimage": "0123456789abcdef...", // preimage for the payment, optional
+        "expiry": 3600, // invoice expiry in seconds, optional
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "make_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": { // result, object. null in case of error.
+        "invoice": "lnbc50n1...",
+        "payment_hash": "31afdf1.."
+    }
+}
+```
+
+### Lookup Invoice
+
+Request:
+```jsonc
+{
+    "method": "lookup_invoice",
+    "params": {
+        "payment_hash": "31afdf1..", // payment hash of the invoice, one of payment_hash or invoice is required
+        "invoice": "lnbc50n1..." // invoice to lookup
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "lookup_invoice",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": {
+        "invoice": "lnbc50n1...", // the encoded invoice
+        "paid": true // true if the invoice has been paid
+    }
+}
+```
+
+
+### List Invoices
+
+Request:
+```jsonc
+{
+    "method": "list_invoices",
+    "params": {
+        "from": 1693876973, // starting timestamp in seconds since epoch, optional
+        "until": 1693876973, // ending timestamp in seconds since epoch, optional
+        "limit": 10, // maximum number of invoices to return, optional
+        "offset": 0, // offset of the first invoice to return, optional
+        "unpaid": true // include unpaid invoices, optional, default false
+    }
+}
+```
+
+Response:
+```jsonc
+{
+    "result_type": "list_invoices",
+    "error": {
+        "code": "UNAUTHORIZED",
+        "message": "human readable error message"
+    },
+    "result": [
+        {
+           "invoice": "string", // encoded invoice
+           "description": "string",
+           "description_hash": "string",
+           "preimage": "string",
+           "payment_hash": "string",
+           "amount": 123, // value in msats
+           "fees_paid": 123, // value in msats
+           "settled_at": unixtimestamp,
+           "metadata": {} // generic metadata that can be used to add things like zap/boostagram details for a payer name/comment/etc.
+        }
+    ],
+}
+```
```

Comment:
The original intention was to be able to create invoices over NWC. That way NWC can be used as a replacement for things like LNURL in the future.

The alby guys wanted to add full functionality to be able to get all other payment information so we added list payments and invoices.

Reply:
> > The original intention was to be able to create invoices over NWC. That way NWC can be used as a replacement for things like LNURL in the future.
> 
> This makes no sense to me, like you would need a separate protocol anyway for zaps that fetch invoices over Nostr since NWC is meant for mostly spending.

Adding a `get_invoice()` command to NWC seems simpler than creating a whole new protocol to do the same thing

Reply:
> This makes no sense to me, like you would need a separate protocol anyway for zaps that fetch invoices over Nostr since NWC is meant for mostly spending.

We discussed this at the beginning in the comments. It makes no sense to connect your wallet but not be able to receive.